File: EcoJardimPro - FronEnd/ecojardimprofrontend/src/models/orcamento/Orcamento.ts
import Projeto from "../projeto/Projeto";
import Servico from "../servico/Servico";

export default class Orcamento {
    public id : number;
    public dataCriacao: string
    public descricao: string
    public projetoId: number
    public projeto:Projeto
    public servicos: Servico[] = [];
    constructor() {
        this.id = 0
        this.dataCriacao = '0001-01-01T00:00:00'
        this.descricao = ''
        this.projetoId = 0
        this.projeto = new Projeto()
        this.servicos = []
    }

}

File: EcoJardimPro - BackEnd/EcoJardimProject/EcoJardimProject.Domain/Entities/Orcamento.cs
using System.ComponentModel.DataAnnotations;
using EcoJardimProject.Domain.Base;

namespace EcoJardimProject.Domain.Entities;

public class Orcamento : IEntity
{
    public long Id { get; set; }
    public string Nome {get;set;}
    [DisplayFormat(DataFormatString = "{0:MMddyyyy}", ApplyFormatInEditMode = true)]
    public DateTime DataCriacao { get; set; }
    public string Descricao { get; set; }
    public long ProjetoId { get; set; }
    public Projeto? Projeto { get; set; }
    public ICollection<Servico> Servicos { get; set; }
    
    public Orcamento()
    {
        Servicos = new List<Servico>();
        DataCriacao = DateTime.Now;
    }
}